src/Microsoft.ML.Data/Transforms/ColumnBindingsBase.cs

```diff
@@ -14,16 +14,40 @@
 
 namespace Microsoft.ML.Data
 {
-    internal abstract class SourceNameColumnBase
+    public class OneToOneColumn
     {
+        /// <summary>Name of the column resulting from the transformation of <see cref="InputColumnName"/>.</summary>
         [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the new column", ShortName = "name")]
-        public string Name;
+        public string OutputColumnName;
+
+        /// <summary>Name of the column to transform. If set to <see langword="null"/>, the value of the <see cref= "OutputColumnName"/> will be used as source.</summary>
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the source column", ShortName = "source, src")]
+        public string InputColumnName = null;
 
-        [Argument(ArgumentType.AtMostOnce, HelpText = "Name of the source column", ShortName = "src")]
-        public string Source;
+        /// <summary>
+        /// Instantiates a <see cref="ColumnOptions"/> from a tuple of input and output column names.
+        /// </summary>
+        public static implicit operator OneToOneColumn((string outputColumnName, string inputColumnName) value)
+        {
+            return new OneToOneColumn(value.outputColumnName, value.inputColumnName);
+        }
+
+        /// <summary>
+        /// Instantiates a <see cref="ColumnOptions"/> from the column name.
+        /// </summary>
+        public static implicit operator OneToOneColumn(string outputColumnName)
```

Comment:
>implicit [](start = 22, length = 8)

I get a little nervous when I see implicit operators from common types, especially since in the same APIs that we're going to be using these sorts of types we're going to have `string` parameters being used. We could sort of get away with it with the value-tuples above -- we don't use value-tuples anywhere else -- but this makes me very nervous.

Reply:
Ok will remove the implicit operators, and simply have a constructor in the options class that will take tuples of strings. 

---
In reply to: [267545690](https://github.com/dotnet/machinelearning/pull/3033#discussion_r267545690) [](ancestors = 267545690)